app/src/main/java/com/tuananh/app/calculatorapp/Calculator.kt

```diff
@@ -0,0 +1,74 @@
+package com.tuananh.app.calculatorapp
+
+import java.util.Stack
+class Calculator {
+
+    companion object {
+        private const val multiply = "*"
+        private const val divide = "/"
+        private const val add = "+"
+        private const val subtract = "-"
+        private const val leftParenthesis = "("
+        private const val rightParenthesis = ")"
+    }
+
+    private val valueStack: Stack<Double> = Stack()
+    private val operatorStack: Stack<String> = Stack()
+
+    fun cal(expression: String): Double {
+        var tokens = expression.trim().split(" ")
+        for(token in tokens) {
+            if(token == "") continue
+            if(token.isDigit()) {
+                valueStack.push(token.toDouble())
+            }
+            else if(token == leftParenthesis) {
+                operatorStack.push(token)
+            }
+            else if(token == rightParenthesis) {
+                while(operatorStack.peek() != leftParenthesis) {
+                    applyOperatorAndPushValueToStack()
+                }
+                operatorStack.pop()
+            }
+
+            //token == operator
```

Comment:
bỏ comment

Reply:
Sao lại phải bỏ ạ?. Comment có vấn đề hay bắt buộc phải bỏ comment ạ?

Reply:
Commit lên thì người ta sẽ ko comment thế này 
```suggestion

            /**
             * comment...
             */
```

Reply:
![image](https://user-images.githubusercontent.com/53604984/127822357-e020b965-3c18-4435-b8d1-826fd798e52c.png)